### PR TITLE
[Amazon AWS] added Step Functions command

### PR DESCRIPTION
**Pull request by nagauta**
## Description

added Step Functions command to search states

## Screencast
スクリーンショット 2024-02-18 1 25 33

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/amazon-aws/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Amazon AWS Changelog
 
+## [Step Functions command] - 2024-02-18
+
+- Added new command to list SQS queues ([#10801](https://github.com/raycast/extensions/pull/10801))
```
Comment (JonathanWbn) [inline extensions/amazon-aws/CHANGELOG.md:5]:
```suggestion
- Added new command to find and open Step Functions ([#10801](https://github.com/raycast/extensions/pull/10801))
```

File: extensions/amazon-aws/src/step-functions.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import { StateMachineListItem, SFNClient, ListStateMachinesCommand } from "@aws-sdk/client-sfn"; // ES Modules import
+import { Action, ActionPanel, Icon, List } from "@raycast/api";
+import { useCachedPromise } from "@raycast/utils";
+import AWSProfileDropdown from "./components/searchbar/aws-profile-dropdown";
+import { resourceToConsoleLink } from "./util";
+
+export default function stepFunctions() {
```
Comment (JonathanWbn) [inline extensions/amazon-aws/src/step-functions.tsx:7]: Not important, but it's a convention to uppercase all React Components, so we could do that here, too.

```suggestion
export default function StepFunctions() {
```

File: extensions/amazon-aws/src/step-functions.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      ) : (
+        stateMachineListItems?.map((stateMachineListItem) => (
+          <StepFunctions key={stateMachineListItem.name} stateMachineListItem={stateMachineListItem} />
+        ))
+      )}
+    </List>
+  );
+}
+
+function StepFunctions({ stateMachineListItem }: { stateMachineListItem: StateMachineListItem }) {
```
Comment (JonathanWbn) [inline extensions/amazon-aws/src/step-functions.tsx:26]:
```suggestion
function StepFunction({ stateMachineListItem }: { stateMachineListItem: StateMachineListItem }) {
```

**Review by JonathanWbn — COMMENTED** [review]
Nice, really cool addition 👍

File: extensions/amazon-aws/src/step-functions.tsx (hunk 2 above)
Comment (nagauta) [reply extensions/amazon-aws/src/step-functions.tsx]: thanks! It sounds great so I renamed!

File: extensions/amazon-aws/src/step-functions.tsx (hunk 3 above)
Comment (nagauta) [reply extensions/amazon-aws/src/step-functions.tsx]: thanks! It sounds great so I renamed!

File: extensions/amazon-aws/CHANGELOG.md (hunk 1 above)
Comment (nagauta) [reply extensions/amazon-aws/CHANGELOG.md]: oh my goodness, that was my bad copy&paste. Thanks, I fixed!

**Comment by nagauta** [comment]
Hi, @JonathanWbn 
Thank you for kind review!! I updated codes, I'm so happy if you check them again😆

**Review by JonathanWbn — APPROVED** [review]
Looks great 🚀

**Review by pernielsentikaer — APPROVED** [review]

**Pull request merged**